src/main/java/dev/hephaestus/glowcase/networking/packet/EditItemDisplayBlockSettings.java

```diff
@@ -0,0 +1,72 @@
+package dev.hephaestus.glowcase.networking.packet;
+
+import dev.hephaestus.glowcase.Glowcase;
+import dev.hephaestus.glowcase.block.entity.ItemDisplayBlockEntity;
+import dev.hephaestus.glowcase.util.NetworkingUtil;
+import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking;
+import net.fabricmc.fabric.api.networking.v1.ServerPlayNetworking;
+import net.minecraft.network.RegistryByteBuf;
+import net.minecraft.network.codec.PacketCodec;
+import net.minecraft.network.codec.PacketCodecs;
+import net.minecraft.network.packet.CustomPayload;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.state.property.Properties;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RotationPropertyHelper;
+
+public record EditItemDisplayBlockSettings(BlockPos pos, ItemDisplayBlockEntity.RotationType rotationType,
+                                           ItemDisplayBlockEntity.GivesItem givesItem,
+                                           ItemDisplayBlockEntity.Offset offset,
+                                           ItemDisplayBlockValues values) implements CustomPayload {
+
+	public static final PacketCodec<RegistryByteBuf, EditItemDisplayBlockSettings> PACKET_CODEC = PacketCodec.tuple(
+		BlockPos.PACKET_CODEC, EditItemDisplayBlockSettings::pos,
+		PacketCodecs.BYTE.xmap(index -> ItemDisplayBlockEntity.RotationType.values()[index], rotation -> (byte) rotation.ordinal()), EditItemDisplayBlockSettings::rotationType,
+		PacketCodecs.BYTE.xmap(index -> ItemDisplayBlockEntity.GivesItem.values()[index], givesItem -> (byte) givesItem.ordinal()), EditItemDisplayBlockSettings::givesItem,
+		PacketCodecs.BYTE.xmap(index -> ItemDisplayBlockEntity.Offset.values()[index], offset -> (byte) offset.ordinal()), EditItemDisplayBlockSettings::offset,
+		ItemDisplayBlockValues.PACKET_CODEC, EditItemDisplayBlockSettings::values,
+		EditItemDisplayBlockSettings::new
+	);
+
+	public static final Id<EditItemDisplayBlockSettings> PACKET_ID = new Id<>(Glowcase.id("channel.item_display"));
+
+	@Override
+	public Id<? extends CustomPayload> getId() {
+		return PACKET_ID;
+	}
+
+	public void send() {
+		ClientPlayNetworking.send(this);
+	}
```

Comment:
can we extract send and receive into `interface C2SPayload extends CustomPayload`? send can be defaulted then.